component/src/main/java/com/synopsys/integration/alert/component/settings/proxy/database/accessor/SettingsProxyConfigAccessor.java

```diff
@@ -0,0 +1,154 @@
+        savedProxyConfig = settingsProxyConfigurationRepository.getOne(savedProxyConfig.getConfigurationId());
+
+        return createConfigModel(savedProxyConfig);
+    }
+
+    @Override
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void deleteConfiguration(UUID configurationId) {
+        if (null != configurationId) {
+            settingsProxyConfigurationRepository.deleteById(configurationId);
+        }
+    }
+
+    private SettingsProxyModel createConfigModel(SettingsProxyConfigurationEntity proxyConfiguration) {
+        SettingsProxyModel newModel = new SettingsProxyModel();
+        String createdAtFormatted = "";
+        String lastUpdatedFormatted = "";
+        if (null != proxyConfiguration) {
+            createdAtFormatted = DateUtils.formatDate(proxyConfiguration.getCreatedAt(), DateUtils.UTC_DATE_FORMAT_TO_MINUTE);
+            if (null != proxyConfiguration.getLastUpdated()) {
+                lastUpdatedFormatted = DateUtils.formatDate(proxyConfiguration.getLastUpdated(), DateUtils.UTC_DATE_FORMAT_TO_MINUTE);
+            }
+            newModel.setProxyHost(proxyConfiguration.getHost());
+            newModel.setProxyPort(proxyConfiguration.getPort());
+            newModel.setProxyUsername(proxyConfiguration.getUsername());
+            if (StringUtils.isNotBlank(proxyConfiguration.getPassword())) {
+                newModel.setProxyPassword(encryptionUtility.decrypt(proxyConfiguration.getPassword()));
+            }
+            newModel.setNonProxyHosts(getNonProxyHosts(proxyConfiguration.getNonProxyHosts()));
+        }
+        newModel.setId(String.valueOf(proxyConfiguration.getConfigurationId()));
+        newModel.setCreatedAt(createdAtFormatted);
+        newModel.setLastUpdated(lastUpdatedFormatted);
+
+        return newModel;
+    }
+
+    private List<String> getNonProxyHosts(List<NonProxyHostConfigurationEntity> nonProxyHosts) {
+        return nonProxyHosts
+                   .stream()
+                   .map(NonProxyHostConfigurationEntity::getHostnamePattern)
+                   .collect(Collectors.toList());
+    }
+
+    private SettingsProxyConfigurationEntity toEntity(UUID configurationId, SettingsProxyModel configuration, OffsetDateTime createdTime, OffsetDateTime lastUpdated) {
+        String host = configuration.getProxyHost().orElseThrow(null);
+        Integer port = configuration.getProxyPort().orElse(null);
+        String username = configuration.getProxyUsername().orElse(null);
+        String password = configuration.getProxyPassword().map(encryptionUtility::encrypt).orElse(null);
+
+        return new SettingsProxyConfigurationEntity(configurationId, createdTime, lastUpdated, host, port, username, password, List.of());
+    }
+
+    private List<NonProxyHostConfigurationEntity> toNonProxyHostEntityList(UUID configurationId, SettingsProxyModel configuration) {
+        List<String> hostnamePatterns = configuration.getNonProxyHosts().orElse(List.of());
```

Comment:
This initial call can be converted into stream mapping that does everything below.